test/models/episode_test.rb

```diff
@@ -166,13 +166,15 @@
     assert_includes episode.keyword_xid, "241 its a title yay"
   end
 
-  it "strips non-alphanumeric characters from all keywords" do
-    episode.update(keywords: ["Here's John,ny!?"])
-    episode.run_callbacks :save do
-      episode.save
-    end
-    refute_nil episode.reload.keywords
-    episode.keywords.each { |k| refute_match(/['?!,]/, k) }
+  it "sanitizes categories" do
+    e = build_stubbed(:episode)
+
+    e.categories = []
+    assert_equal [], e.categories
+    assert_nil e[:categories]
+
+    e.categories = ["foo", " Foo ", "BAR  ", "  foo", "!@  $?"]
+    assert_equal ["foo", "Foo", "BAR", "!@  $?"], e.categories
```

Comment:
oh it doesn't do case?

Reply:
Do we want it to?  That may not be a bad idea.  De-dup based on whichever is first in the array.

Reply:
I feel like there is no good reason to have Foo and foo, or "Arts" and "arts" if that is a better case.

Reply:
It would probably be good to prevent this at the time of entry in the UI

Reply:
Okay, added this.  And this did trim down on dup categories on a bunch of episodes (where they had both `keywords` and `categories` set ... i made the migration prefer the non-lowercased `categories`).

Reply:
Opened #1018 for the UI bit.